File: src/components/Leaderboard.js
import React, { useState, useEffect } from 'react';

import './Leaderboard.css';

function Leaderboard(props) {
  const [rankings, setRankings] = useState([]);

  useEffect(() => {
    fetch('/api/leaderboard/list').then((response) => {
      response.json().then((data) => setRankings(data.ranks));
    });
  }, []);

  return (
    <div>
      <table>
        <thead>
          <tr>
            <td className="tableTitle">
              <b> Rank </b>
            </td>
            <td className="tableTitle">
              <b> Name </b>
            </td>
            <td className="tableTitle">
              <b> Score </b>
            </td>
          </tr>
        </thead>
        <tbody>
          {rankings.map((entry) => (
            <tr key={entry.name}>
              <td>{entry.rank}</td>
              <td>{entry.name}</td>
              <td>{entry.points}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default Leaderboard;

File: server/routes/leaderboard.py
from flask import Blueprint, session


leaderboard_bp = Blueprint("leaderboard_bp", __name__, url_prefix="/api/leaderboard")


@leaderboard_bp.route("/list")
def leaderboard():
    data = [
        {"rank": 1, "name": "Naman", "points": 120},
        {"rank": 2, "name": "Anshul", "points": 88},
        {"rank": 3, "name": "Kris", "points": 60},
        {"rank": 4, "name": "David", "points": 0},
    ]
    return {"ranks": data}
